solr/solrj/src/java/org/apache/solr/common/util/JsonSchemaValidator.java

```diff
@@ -48,13 +48,15 @@ public JsonSchemaValidator(String jsonString) {
 
   public JsonSchemaValidator(Map jsonSchema) {
     this.validators = new LinkedList<>();
-    for (Object fname : jsonSchema.keySet()) {
+    for (Object entry : jsonSchema.entrySet()) {
```

Comment:
use Map.Entry<> directly here

Reply:
+1 (same for the other loop below)

Reply:
I've got to admit I'm a bit confused on this since `Map.Entry<>` is not what one gets from `entrySet` of a `Map`:

```
> Task :solr:solrj:compileJava
/code/solr/solrj/src/java/org/apache/solr/common/util/JsonSchemaValidator.java:51: error: incompatible types: Object ca
nnot be converted to Entry<?,?>
    for (Map.Entry<?,?> entry : jsonSchema.entrySet()) {
```

Are you asking for us to change the function from:

```
    public JsonSchemaValidator(Map jsonSchema) {
```

to

```
    public JsonSchemaValidator(Map<?,?> jsonSchema) {
```

And propagate that API change to the callees, or did I miss the obvious thing?

Reply:
I was being foolish - Map<> is abstract not concrete and we have no problem.  Never mind, will update.